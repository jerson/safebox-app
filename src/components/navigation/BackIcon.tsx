import React from 'react';
import {
  Image,
  ImageStyle,
  StyleProp,
  StyleSheet,
  TouchableOpacity,
  ViewStyle
} from 'react-native';
import Colors from '../../modules/constants/Colors';

const styles = StyleSheet.create({
  back: {
    width: 21,
    height: 21,
    tintColor: Colors.grey4
  } as ImageStyle,
  container: {
    height: 50,
    width: 50,
    paddingLeft: 20,
    justifyContent: 'center',
    backgroundColor: 'transparent',
    opacity: 1
  } as ViewStyle
});

export interface BackIconProps {
  onPress?: () => void;
  style?: StyleProp<ImageStyle>;
}

const BackIcon: React.FC<BackIconProps> = ({ style, onPress, ...props }) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <Image
        style={[styles.back, style]}
        source={require('../../assets/images/back-icon.png')}
        {...props}
      />
    </TouchableOpacity>
  );
};

export default BackIcon;
