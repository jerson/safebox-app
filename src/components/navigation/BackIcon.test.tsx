import React from 'react';

import * as renderer from 'react-test-renderer';
import BackIcon, { BackIconProps } from './BackIcon';

const props: BackIconProps = {};

it('renders without crashing', () => {
  const snapshot = renderer.create(<BackIcon {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

it('render just views', () => {
  const snapshot = renderer.create(<BackIcon {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
