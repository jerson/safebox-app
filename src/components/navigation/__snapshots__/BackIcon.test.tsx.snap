// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "paddingLeft": 20,
      "width": 50,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/back-icon.png",
      }
    }
    style={
      Array [
        Object {
          "height": 21,
          "tintColor": "#d2def1",
          "width": 21,
        },
        undefined,
      ]
    }
  />
</View>
`;
